api/v1alpha1/kongplugin_installation_types.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2022 Kong Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&KongPluginInstallation{}, &KongPluginInstallationList{})
+}
+
+//+genclient
+//+k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
+//+kubebuilder:resource:shortName=kpi,categories=kong;all
+//+kubebuilder:subresource:status
+//+kubebuilder:printcolumn:name="Ready",type=string,JSONPath=`.status.conditions[?(@.type=="Ready")].status`
+
+// KongPluginInstallation allows using a custom Kong Plugin distributed as a container image available in a registry.
+// Such a plugin can be associated with GatewayConfiguration or DataPlane to be available for particular Kong Gateway
+// and configured with KongPlugin CRD.
+type KongPluginInstallation struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   KongPluginInstallationSpec   `json:"spec,omitempty"`
+	Status KongPluginInstallationStatus `json:"status,omitempty"`
+}
+
+//+kubebuilder:object:root=true
+
+// KongPluginInstallationList contains a list of KongPluginInstallation.
+type KongPluginInstallationList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata,omitempty"`
+	Items           []KongPluginInstallation `json:"items"`
+}
+
+// KongPluginInstallationSpec defines the desired state of KongPluginInstallation.
+type KongPluginInstallationSpec struct {
+
+	// The image is an OCI image URL for a packaged Custom Kong Plugin.
```

Comment:
```suggestion
	// The image is an OCI image URL for a packaged custom Kong plugin.
```
https://docs.konghq.com/gateway/latest/kong-plugins/#custom-plugins doesn't capitalize these terms, we shouldn't either.